In [1]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/user-4-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           120673071          NaN   
1           1                19906           120692977  Eye Tracker   
2           2                76745           120749816  Eye Tracker   
3           3                93388           120766459  Eye Tracker   
4           4               110076           120783147  Eye Tracker   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-4     Recording5     2025/07/18   
1          JD4  2025/07/18           user-4     Recording5     2025/07/18   
2          JD4  2025/07/18           user-4     Recording5     2025/07/18   
3          JD4  2025/07/18           user-4     Recording5     2025/07/18   
4          JD4  2025/07/18           user-4     Recording5     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN           Saccade   
4         2025/07/18  ...                   NaN           Saccade   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                         75.0                     1.0              NaN   
2                         75.0                     1.0              NaN   
3                         50.0                     1.0              NaN   
4                         50.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  

[5 rows x 84 columns]

# カラムの整理

In [3]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    10873
Recording timestamp                           10860
Computer timestamp                            10860
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [4]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [5]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-4']
Recording name: ['Recording5']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:30:57.621']
Recording start time UTC: ['07:30:57.621']
Recording duration: [194658]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [22.7]
Average calibration accuracy (degrees): [1.59]
Average calibration accuracy (pixels): [79]
Average validation accuracy (mm): [17.4]
Average validation precision SD (mm): [5.1]
Average validation precision RMS (mm): [7.2]
Average validation accuracy (degrees): [1.26]
Average validation precision SD (degrees): [0.37]
Average validation precision RMS (degrees): [0.52]
Average validation accuracy (pixels): [61]
Average validation precision

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-4`|
|Recording name| 録画名|`Recording5`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:30:57.621`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:30:57.621`|
|Recording duration| 録画時間|`194658`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`22.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.59`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`79`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.4`mm|
|Average validation precision SD | 平均バリデーション予測SD|`5.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`7.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.26`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.37`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.52`degrees|
|Average validation accuracy |平均バリデーション精度| `61`pixels|
|Average validation precision SD |平均バリデーション予測SD| `18`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `25`pixels|

In [6]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           120673071          NaN   
1           1                19906           120692977  Eye Tracker   
2           2                76745           120749816  Eye Tracker   
3           3                93388           120766459  Eye Tracker   
4           4               110076           120783147  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1           136085195.0             NaN         NaN           NaN   
2           136142062.0             NaN         NaN           NaN   
3           136158705.0             NaN         NaN        1504.0   
4           136175391.0             NaN         NaN        1549.0   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3         935.0             1574.0  ...                    NaN   
4         926.0             1624.0  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                         75.0                      1.0   
2       EyesNotFound                         75.0                      1.0   
3            Saccade                         50.0                      1.0   
4            Saccade                         50.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN               NaN               NaN  
2                         NaN               NaN               NaN  
3                         NaN               NaN               NaN  
4                         NaN               NaN               NaN  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [7]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             10873
Recording timestamp                    10860
Computer timestamp                     10860
Sensor                                     3
Eyetracker timestamp                   10487
Event                                      9
Event value                                9
Gaze point X                            1757
Gaze point Y                             889
Gaze point left X                       1738
Gaze point left Y                        893
Gaze point right X                      1750
Gaze point right Y                       895
Gaze direction left X                   8741
Gaze direction left Y                   8021
Gaze direction left Z                   4735
Gaze direction right X                  8388
Gaze direction right Y                  7534
Gaze direction right Z                  4726
Pupil diameter left                     1325
Pupil diameter right                    1110
Pupil diameter filtered                 1205
Validity l

## Sensor事にデータフレームを分割

In [8]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [9]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0           120673071    NaN   
240           240              2943204           123616275    NaN   
668           668              7402612           128075683    NaN   
676           676              7508837           128181908    NaN   
2053         2053             47305117           167978188    NaN   
2059         2059             47421178           168094249    NaN   
2067         2067             47786220           168459291    NaN   
2068         2068             47786220           168459291    NaN   
2392         2392             52810973           173484044    NaN   
2393         2393             52810973           173484044    NaN   
4178         4178             82497394           203170465    NaN   
4181         4181             82540194           203213265    NaN   
4182         4182             82540194           203213265    NaN   
4483         4483             87534774           208207845    NaN   
4484         4484             87534774           208207845    NaN   
6174         6174            115725406           236398477    NaN   
6177         6177            115762236           236435307    NaN   
6178         6178            115762236           236435307    NaN   
6479         6479            120752839           241425910    NaN   
6480         6480            120752839           241425910    NaN   
8356         8356            151956013           272629084    NaN   
8359         8359            151984304           272657375    NaN   
8360         8360            151984304           272657375    NaN   
8662         8662            156993031           277666102    NaN   
8663         8663            156993031           277666102    NaN   
10609       10609            189403330           310076401    NaN   
10613       10613            189443114           310116185    NaN   
10614       10614            189443114           310116185    NaN   
10869       10869            194434180           315107251    NaN   
10872       10872            194658164           315331235    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
240                     NaN  Eye tracker Calibration start         NaN   
668                     NaN                     MouseEvent  Down, Left   
676                     NaN                     MouseEvent    Up, Left   
2053                    NaN                     MouseEvent  Down, Left   
2059                    NaN                     MouseEvent    Up, Left   
2067                    NaN    Eye tracker Calibration end         NaN   
2068                    NaN             ImageStimulusStart      blank2   
2392                    NaN               ImageStimulusEnd      blank2   
2393                    NaN             ImageStimulusStart  視線誘導問題.002   
4178                    NaN                  KeyboardEvent       Space   
4181                    NaN               ImageStimulusEnd  視線誘導問題.002   
4182                    NaN             ImageStimulusStart      blank2   
4483                    NaN               ImageStimulusEnd      blank2   
4484                    NaN             ImageStimulusStart  視線誘導問題.001   
6174                    NaN                  KeyboardEvent       Space   
6177                    NaN               ImageStimulusEnd  視線誘導問題.001   
6178                    NaN             ImageStimulusStart      blank2   
6479                    NaN               ImageStimulusEnd      blank2   
6480                    NaN             ImageStimulusStart  視線誘導問題.004   
8356                    NaN                  KeyboardEvent       Space   
8359                    NaN               ImageStimulusEnd  視線誘導問題.004   
8360                    NaN             ImageStimulusStart      blank2   
8662                    NaN               ImageStimulusEnd      blank2   
8663             

In [10]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
2393              52810973           173484044  視線誘導問題.002
4181              82540194           203213265  視線誘導問題.002
4484              87534774           208207845  視線誘導問題.001
6177             115762236           236435307  視線誘導問題.001
6480             120752839           241425910  視線誘導問題.004
8359             151984304           272657375  視線誘導問題.004
8663             156993031           277666102  視線誘導問題.003
10613            189443114           310116185  視線誘導問題.003


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|7057|131493532|6790447553|視線誘導問題.001|
|8983|163520969|6822474990|視線誘導問題.001|
|9287|168529652|6827483673|視線誘導問題.004|
|11198|200288917|6859242938|視線誘導問題.004|
|11502|205296898|6864250919|視線誘導問題.002|
|13757|242247315|6901201336|視線誘導問題.002|
|14047|247255399|6906209420|視線誘導問題.003|
|16047|280802867|6939756888|視線誘導問題.003|

In [11]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/start-stop-timing.csv")

In [12]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
33             33               581632           121254703  Mouse   
37             37               628241           121301312  Mouse   
39             39               652087           121325158  Mouse   
42             42               684339           121357410  Mouse   
46             46               738137           121411208  Mouse   
...           ...                  ...                 ...    ...   
10603       10603            189361237           310034308  Mouse   
10604       10604            189375128           310048199  Mouse   
10606       10606            189389065           310062136  Mouse   
10608       10608            189403019           310076090  Mouse   
10610       10610            189410078           310083149  Mouse   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
33                      NaN   NaN         NaN           NaN           NaN   
37                      NaN   NaN         NaN           NaN           NaN   
39                      NaN   NaN         NaN           NaN           NaN   
42                      NaN   NaN         NaN           NaN           NaN   
46                      NaN   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
10603                   NaN   NaN         NaN           NaN           NaN   
10604                   NaN   NaN         NaN           NaN           NaN   
10606                   NaN   NaN         NaN           NaN           NaN   
10608                   NaN   NaN         NaN           NaN           NaN   
10610                   NaN   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
33                   NaN  ...                    NaN            Saccade   
37                   NaN  ...                    NaN           Fixation   
39                   NaN  ...                    NaN           Fixation   
42                   NaN  ...                    NaN           Fixation   
46                   NaN  ...                    NaN           Fixation   
...                  ...  ...                    ...                ...   
10603                NaN  ...                 1080.0       EyesNotFound   
10604                NaN  ...                 1080.0       EyesNotFound   
10606                NaN  ...                 1080.0       EyesNotFound   
10608                NaN  ...                 1080.0       EyesNotFound   
10610                NaN  ...                 1080.0       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
33                            50.0                      3.0               NaN   
37                           183.0                      3.0             656.0   
39                           183.0                      3.0             656.0   
42                           183.0                      3.0             656.0   
46                           183.0                      3.0             656.0   
...                            ...                      ...               ...   
10603                       4176.0                    110.0               NaN   
10604                       4176.0                    110.0               NaN   
10606                       4176.0                    110.0               NaN   
10608                       4176.0                    110.0               NaN   
10610                       4176.0                    110.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
33                  NaN                         NaN   
37                411.0                         NaN   
39                411.0                         NaN   
42                411.0                         NaN   
46                411.0                         NaN   
...                 ...                         .

In [13]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                19906           120692977  Eye Tracker   
2               2                76745           120749816  Eye Tracker   
3               3                93388           120766459  Eye Tracker   
4               4               110076           120783147  Eye Tracker   
5               5               126878           120799949  Eye Tracker   
...           ...                  ...                 ...          ...   
10866       10866            194358659           315031730  Eye Tracker   
10867       10867            194398601           315071672  Eye Tracker   
10868       10868            194429862           315102933  Eye Tracker   
10870       10870            194446148           315119219  Eye Tracker   
10871       10871            194486488           315159559  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1               136085195.0   NaN         NaN           NaN           NaN   
2               136142062.0   NaN         NaN           NaN           NaN   
3               136158705.0   NaN         NaN        1504.0         935.0   
4               136175391.0   NaN         NaN        1549.0         926.0   
5               136192188.0   NaN         NaN        1915.0         831.0   
...                     ...   ...         ...           ...           ...   
10866           330412112.0   NaN         NaN           NaN           NaN   
10867           330452052.0   NaN         NaN           NaN           NaN   
10868           330483311.0   NaN         NaN           NaN           NaN   
10870           330499596.0   NaN         NaN           NaN           NaN   
10871           330539934.0   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
2                    NaN  ...                    NaN       EyesNotFound   
3                 1574.0  ...                    NaN            Saccade   
4                 1624.0  ...                    NaN            Saccade   
5                 2052.0  ...                    NaN            Saccade   
...                  ...  ...                    ...                ...   
10866                NaN  ...                 1080.0       EyesNotFound   
10867                NaN  ...                 1080.0       EyesNotFound   
10868                NaN  ...                 1080.0       EyesNotFound   
10870                NaN  ...                    NaN       EyesNotFound   
10871                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                             75.0                      1.0               NaN   
2                             75.0                      1.0               NaN   
3                             50.0                      1.0               NaN   
4                             50.0                      1.0               NaN   
5                             50.0                      1.0               NaN   
...                            ...                      ...               ...   
10866                       2022.0                    117.0               NaN   
10867                       2022.0                    117.0               NaN   
10868                       2022.0                    117.0               NaN   
10870                       2022.0                    117.0               NaN   
10871                       2022.0                    117.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
2                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN         

## Mouse以外のデータを抽出

In [14]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [15]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [16]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             10516
Recording timestamp                    10507
Computer timestamp                     10507
Sensor                                     2
Eyetracker timestamp                   10487
Event                                      9
Event value                                9
Gaze point X                            1757
Gaze point Y                             889
Gaze point left X                       1738
Gaze point left Y                        893
Gaze point right X                      1750
Gaze point right Y                       895
Gaze direction left X                   8741
Gaze direction left Y                   8021
Gaze direction left Z                   4735
Gaze direction right X                  8388
Gaze direction right Y                  7534
Gaze direction right Z                  4726
Pupil diameter left                     1325
Pupil diameter right                    1110
Pupil diameter filtered                 1205
Validity l

In [17]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           120673071          NaN   
1           1                19906           120692977  Eye Tracker   
2           2                76745           120749816  Eye Tracker   
3           3                93388           120766459  Eye Tracker   
4           4               110076           120783147  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1           136085195.0             NaN         NaN           NaN   
2           136142062.0             NaN         NaN           NaN   
3           136158705.0             NaN         NaN        1504.0   
4           136175391.0             NaN         NaN        1549.0   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
1           NaN                NaN  ...                                  NaN   
2           NaN                NaN  ...                                  NaN   
3         935.0             1574.0  ...                                  NaN   
4         926.0             1624.0  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
1                   NaN                    NaN       EyesNotFound   
2                   NaN                    NaN       EyesNotFound   
3                   NaN                    NaN            Saccade   
4                   NaN                    NaN            Saccade   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
1                         75.0                      1.0               NaN   
2                         75.0                      1.0               NaN   
3                         50.0                      1.0               NaN   
4                         50.0                      1.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
1               NaN                         NaN                         NaN  
2               NaN                         NaN                         NaN  
3               NaN                         NaN                         NaN  
4               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [18]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                19906           120692977  Eye Tracker   
2               2                76745           120749816  Eye Tracker   
3               3                93388           120766459  Eye Tracker   
4               4               110076           120783147  Eye Tracker   
5               5               126878           120799949  Eye Tracker   
...           ...                  ...                 ...          ...   
10866       10866            194358659           315031730  Eye Tracker   
10867       10867            194398601           315071672  Eye Tracker   
10868       10868            194429862           315102933  Eye Tracker   
10870       10870            194446148           315119219  Eye Tracker   
10871       10871            194486488           315159559  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1               136085195.0   NaN         NaN           NaN           NaN   
2               136142062.0   NaN         NaN           NaN           NaN   
3               136158705.0   NaN         NaN        1504.0         935.0   
4               136175391.0   NaN         NaN        1549.0         926.0   
5               136192188.0   NaN         NaN        1915.0         831.0   
...                     ...   ...         ...           ...           ...   
10866           330412112.0   NaN         NaN           NaN           NaN   
10867           330452052.0   NaN         NaN           NaN           NaN   
10868           330483311.0   NaN         NaN           NaN           NaN   
10870           330499596.0   NaN         NaN           NaN           NaN   
10871           330539934.0   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
2                    NaN  ...                    NaN       EyesNotFound   
3                 1574.0  ...                    NaN            Saccade   
4                 1624.0  ...                    NaN            Saccade   
5                 2052.0  ...                    NaN            Saccade   
...                  ...  ...                    ...                ...   
10866                NaN  ...                 1080.0       EyesNotFound   
10867                NaN  ...                 1080.0       EyesNotFound   
10868                NaN  ...                 1080.0       EyesNotFound   
10870                NaN  ...                    NaN       EyesNotFound   
10871                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                             75.0                      1.0               NaN   
2                             75.0                      1.0               NaN   
3                             50.0                      1.0               NaN   
4                             50.0                      1.0               NaN   
5                             50.0                      1.0               NaN   
...                            ...                      ...               ...   
10866                       2022.0                    117.0               NaN   
10867                       2022.0                    117.0               NaN   
10868                       2022.0                    117.0               NaN   
10870                       2022.0                    117.0               NaN   
10871                       2022.0                    117.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
2                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN         

## 各問題毎のデータへ切り分け

In [20]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [21]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2394        2394             52822648           173495719  Eye Tracker   
2395        2395             52839878           173512949  Eye Tracker   
2396        2396             52856528           173529599  Eye Tracker   
2397        2397             52872571           173545642  Eye Tracker   
2398        2398             52889810           173562881  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2394           188884778.0   NaN         NaN        1254.0         384.0   
2395           188902007.0   NaN         NaN        1336.0         389.0   
2396           188918656.0   NaN         NaN        1335.0         385.0   
2397           188934698.0   NaN         NaN        1332.0         384.0   
2398           188951936.0   NaN         NaN        1340.0         381.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2394             1249.0  ...                 1080.0            Saccade   
2395             1326.0  ...                 1080.0            Saccade   
2396             1330.0  ...                 1080.0           Fixation   
2397             1336.0  ...                 1080.0           Fixation   
2398             1345.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2394                         84.0                    111.0               NaN   
2395                         84.0                    111.0               NaN   
2396                        166.0                     82.0            1338.0   
2397                        166.0                     82.0            1338.0   
2398                        166.0                     82.0            1338.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
2394               NaN                         NaN   
2395               NaN                         NaN   
2396             381.0                      0.6968   
2397             381.0                      0.6968   
2398             381.0                      0.6968   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
2394                         NaN               NaN               NaN  
2395                         NaN               NaN               NaN  
2396                      0.3526               NaN               NaN  
2397                      0.3526               NaN               NaN  
2398                      0.3526               NaN               NaN  

[5 rows x 57 columns]

In [22]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1786
Recording timestamp                    1786
Computer timestamp                     1786
Sensor                                    1
Eyetracker timestamp                   1786
Event                                     1
Event value                               1
Gaze point X                            896
Gaze point Y                            468
Gaze point left X                       902
Gaze point left Y                       471
Gaze point right X                      848
Gaze point right Y                      375
Gaze direction left X                  1654
Gaze direction left Y                  1537
Gaze direction left Z                  1396
Gaze direction right X                 1495
Gaze direction right Y                 1323
Gaze direction right Z                 1284
Pupil diameter left                     691
Pupil diameter right                    580
Pupil diameter filtered                 624
Validity left                   

In [23]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
4485        4485             87536399           208209470  Eye Tracker   
4486        4486             87553044           208226115  Eye Tracker   
4487        4487             87569087           208242158  Eye Tracker   
4488        4488             87586271           208259342  Eye Tracker   
4489        4489             87602967           208276038  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
4485           223596412.0   NaN         NaN         391.0         547.0   
4486           223613056.0   NaN         NaN         398.0         548.0   
4487           223629098.0   NaN         NaN         390.0         539.0   
4488           223646281.0   NaN         NaN         382.0         537.0   
4489           223662976.0   NaN         NaN         390.0         543.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
4485              382.0  ...                 1080.0           Fixation   
4486              384.0  ...                 1080.0           Fixation   
4487              386.0  ...                 1080.0           Fixation   
4488              388.0  ...                 1080.0           Fixation   
4489              385.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
4485                        283.0                    195.0             392.0   
4486                        283.0                    195.0             392.0   
4487                        283.0                    195.0             392.0   
4488                        283.0                    195.0             392.0   
4489                        283.0                    195.0             392.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
4485             539.0                       0.204   
4486             539.0                       0.204   
4487             539.0                       0.204   
4488             539.0                       0.204   
4489             539.0                       0.204   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
4485                      0.4987               NaN               NaN  
4486                      0.4987               NaN               NaN  
4487                      0.4987               NaN               NaN  
4488                      0.4987               NaN               NaN  
4489                      0.4987               NaN               NaN  

[5 rows x 57 columns]

In [24]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8664        8664            157005027           277678098  Eye Tracker   
8665        8665            157021668           277694739  Eye Tracker   
8666        8666            157037711           277710782  Eye Tracker   
8667        8667            157054951           277728022  Eye Tracker   
8668        8668            157071592           277744663  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8664           293060771.0   NaN         NaN         566.0         550.0   
8665           293077411.0   NaN         NaN         558.0         545.0   
8666           293093453.0   NaN         NaN         552.0         554.0   
8667           293110691.0   NaN         NaN         554.0         556.0   
8668           293127331.0   NaN         NaN         515.0         545.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8664              578.0  ...                 1080.0           Fixation   
8665              575.0  ...                 1080.0           Fixation   
8666              567.0  ...                 1080.0           Fixation   
8667              561.0  ...                 1080.0           Fixation   
8668              517.0  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8664                        350.0                    442.0             557.0   
8665                        350.0                    442.0             557.0   
8666                        350.0                    442.0             557.0   
8667                        350.0                    442.0             557.0   
8668                         16.0                    637.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
8664             550.0                      0.2903   
8665             550.0                      0.2903   
8666             550.0                      0.2903   
8667             550.0                      0.2903   
8668               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8664                      0.5097               NaN               NaN  
8665                      0.5097               NaN               NaN  
8666                      0.5097               NaN               NaN  
8667                      0.5097               NaN               NaN  
8668                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [25]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
6481        6481            120760424           241433495  Eye Tracker   
6482        6482            120777516           241450587  Eye Tracker   
6483        6483            120793564           241466635  Eye Tracker   
6484        6484            120810393           241483464  Eye Tracker   
6485        6485            120827444           241500515  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
6481           256818395.0   NaN         NaN         774.0         617.0   
6482           256835486.0   NaN         NaN         617.0         603.0   
6483           256851533.0   NaN         NaN         520.0         577.0   
6484           256868361.0   NaN         NaN         508.0         586.0   
6485           256885411.0   NaN         NaN         502.0         584.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
6481              785.0  ...                 1080.0            Saccade   
6482              623.0  ...                 1080.0            Saccade   
6483              530.0  ...                 1080.0            Saccade   
6484              522.0  ...                 1080.0           Fixation   
6485              521.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
6481                         66.0                    436.0               NaN   
6482                         66.0                    436.0               NaN   
6483                         66.0                    436.0               NaN   
6484                        267.0                    306.0             499.0   
6485                        267.0                    306.0             499.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
6481               NaN                         NaN   
6482               NaN                         NaN   
6483               NaN                         NaN   
6484             573.0                        0.26   
6485             573.0                        0.26   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
6481                         NaN               NaN               NaN  
6482                         NaN               NaN               NaN  
6483                         NaN               NaN               NaN  
6484                       0.531               NaN               NaN  
6485                       0.531               NaN               NaN  

[5 rows x 57 columns]

In [26]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presen

In [27]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-4`|
|Recording name| 録画名|`Recording5`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:30:57.621`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:30:57.621`|
|Recording duration| 録画時間|`194658`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`22.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.59`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`79`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.4`mm|
|Average validation precision SD | 平均バリデーション予測SD|`5.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`7.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.26`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.37`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.52`degrees|
|Average validation accuracy |平均バリデーション精度| `61`pixels|
|Average validation precision SD |平均バリデーション予測SD| `18`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `25`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [28]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
4485        4485             87536399           208209470   
4486        4486             87553044           208226115   
4487        4487             87569087           208242158   
4488        4488             87586271           208259342   
4489        4489             87602967           208276038   
...          ...                  ...                 ...   
6171        6171            115685383           236358454   
6172        6172            115701426           236374497   
6173        6173            115718665           236391736   
6175        6175            115735306           236408377   
6176        6176            115751349           236424420   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
4485           223596412.0         391.0         547.0              382.0   
4486           223613056.0         398.0         548.0              384.0   
4487           223629098.0         390.0         539.0              386.0   
4488           223646281.0         382.0         537.0              388.0   
4489           223662976.0         390.0         543.0              385.0   
...                    ...           ...           ...                ...   
6171           251743660.0        1297.0         247.0             1283.0   
6172           251759702.0         956.0         240.0              915.0   
6173           251776940.0         610.0         216.0              610.0   
6175           251793580.0         454.0         233.0              454.0   
6176           251809622.0           NaN           NaN                NaN   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
4485              529.0               400.0               565.0  ...   
4486              525.0               412.0               571.0  ...   
4487              526.0               393.0               551.0  ...   
4488              511.0               375.0               564.0  ...   
4489              502.0               395.0               584.0  ...   
...                 ...                 ...                 ...  ...   
6171              254.0              1311.0               241.0  ...   
6172              229.0               997.0               250.0  ...   
6173              216.0                 NaN                 NaN  ...   
6175              233.0                 NaN                 NaN  ...   
6176                NaN                 NaN                 NaN  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
4485                       0.4895                        0.2083   
4486                       0.4864                        0.2144   
4487                       0.4875                        0.2048   
4488                       0.4733                        0.1956   
4489                       0.4651                        0.2055   
...                           ...                           ...   
6171                       0.2354                        0.6828   
6172                       0.2123                        0.5192   
6173                       0.1996                           NaN   
6175                       0.2161                           NaN   
6176                          NaN                           NaN   

      Gaze point right Y (MCSnorm)  Eye movement type  \
4485                        0.5236           Fixation   
4486                        0.5287           Fixation   
4487                        0.5098           Fixation   
4488                        0.5220           Fixation   
4489                        0.5407           Fixation   
...                            ...                ...   
6171                        0.2227            Saccade   
6172                        0.2318            Saccade   
6173                           NaN            Saccade   
6175                           NaN            Saccade   
6176                           NaN       EyesNotFou

In [29]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/user-4-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-4`|
|Recording name| 録画名|`Recording5`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:30:57.621`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:30:57.621`|
|Recording duration| 録画時間|`194658`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`22.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.59`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`79`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.4`mm|
|Average validation precision SD | 平均バリデーション予測SD|`5.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`7.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.26`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.37`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.52`degrees|
|Average validation accuracy |平均バリデーション精度| `61`pixels|
|Average validation precision SD |平均バリデーション予測SD| `18`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `25`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
6481        6481            120760424           241433495   
6482        6482            120777516           241450587   
6483        6483            120793564           241466635   
6484        6484            120810393           241483464   
6485        6485            120827444           241500515   
...          ...                  ...                 ...   
8353        8353            151912879           272585950   
8354        8354            151929520           272602591   
8355        8355            151945563           272618634   
8357        8357            151962802           272635873   
8358        8358            151979443           272652514   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
6481           256818395.0         774.0         617.0              785.0   
6482           256835486.0         617.0         603.0              623.0   
6483           256851533.0         520.0         577.0              530.0   
6484           256868361.0         508.0         586.0              522.0   
6485           256885411.0         502.0         584.0              521.0   
...                    ...           ...           ...                ...   
8353           287968931.0        1239.0         470.0             1229.0   
8354           287985571.0        1213.0         415.0             1208.0   
8355           288001613.0        1131.0         269.0             1103.0   
8357           288018851.0        1068.0         222.0             1051.0   
8358           288035491.0        1036.0         232.0             1038.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
6481              613.0               763.0               622.0  ...   
6482              613.0               611.0               593.0  ...   
6483              595.0               511.0               559.0  ...   
6484              597.0               494.0               576.0  ...   
6485              604.0               484.0               564.0  ...   
...                 ...                 ...                 ...  ...   
8353              460.0              1248.0               479.0  ...   
8354              417.0              1218.0               412.0  ...   
8355              228.0              1160.0               311.0  ...   
8357              216.0              1085.0               229.0  ...   
8358              218.0              1034.0               247.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
6481                       0.5676                        0.3975   
6482                       0.5679                        0.3181   
6483                       0.5511                        0.2660   
6484                       0.5524                        0.2571   
6485                       0.5594                        0.2521   
...                           ...                           ...   
8353                       0.4260                        0.6501   
8354                       0.3862                        0.6346   
8355                       0.2110                        0.6039   
8357                       0.2000                        0.5649   
8358                       0.2016                        0.5386   

      Gaze point right Y (MCSnorm)  Eye movement type  \
6481                        0.5755            Saccade   
6482                        0.5493            Saccade   
6483                        0.5178            Saccade   
6484                        0.5334           Fixation   
6485                        0.5227           Fixation   
...                            ...                ...   
8353                        0.4438           Fixation   
8354                        0.3816            Saccade   
8355                        0.2876            Saccade   
8357                        0.2119            Saccade   
8358                        0.2283            Sacca

In [31]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/user-4-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-4`|
|Recording name| 録画名|`Recording5`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:30:57.621`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:30:57.621`|
|Recording duration| 録画時間|`194658`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`22.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.59`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`79`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.4`mm|
|Average validation precision SD | 平均バリデーション予測SD|`5.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`7.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.26`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.37`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.52`degrees|
|Average validation accuracy |平均バリデーション精度| `61`pixels|
|Average validation precision SD |平均バリデーション予測SD| `18`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `25`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
2394        2394             52822648           173495719   
2395        2395             52839878           173512949   
2396        2396             52856528           173529599   
2397        2397             52872571           173545642   
2398        2398             52889810           173562881   
...          ...                  ...                 ...   
4175        4175             82460897           203133968   
4176        4176             82476940           203150011   
4177        4177             82494179           203167250   
4179        4179             82510820           203183891   
4180        4180             82526863           203199934   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
2394           188884778.0        1254.0         384.0             1249.0   
2395           188902007.0        1336.0         389.0             1326.0   
2396           188918656.0        1335.0         385.0             1330.0   
2397           188934698.0        1332.0         384.0             1336.0   
2398           188951936.0        1340.0         381.0             1345.0   
...                    ...           ...           ...                ...   
4175           218521216.0        1431.0         259.0             1474.0   
4176           218537258.0        1416.0         263.0             1424.0   
4177           218554496.0        1419.0         254.0             1431.0   
4179           218571136.0        1418.0         258.0             1448.0   
4180           218587178.0        1393.0         268.0                NaN   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
2394              381.0              1259.0               386.0  ...   
2395              376.0              1347.0               402.0  ...   
2396              368.0              1340.0               403.0  ...   
2397              372.0              1327.0               396.0  ...   
2398              373.0              1335.0               389.0  ...   
...                 ...                 ...                 ...  ...   
4175              258.0              1389.0               260.0  ...   
4176              257.0              1408.0               269.0  ...   
4177              251.0              1407.0               257.0  ...   
4179              256.0              1388.0               259.0  ...   
4180                NaN              1393.0               268.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
2394                       0.3528                        0.6556   
2395                       0.3480                        0.7016   
2396                       0.3406                        0.6977   
2397                       0.3440                        0.6910   
2398                       0.3449                        0.6954   
...                           ...                           ...   
4175                       0.2393                        0.7233   
4176                       0.2384                        0.7335   
4177                       0.2321                        0.7327   
4179                       0.2369                        0.7231   
4180                          NaN                        0.7253   

      Gaze point right Y (MCSnorm)  Eye movement type  \
2394                        0.3577            Saccade   
2395                        0.3721            Saccade   
2396                        0.3728           Fixation   
2397                        0.3667           Fixation   
2398                        0.3606           Fixation   
...                            ...                ...   
4175                        0.2405           Fixation   
4176                        0.2489           Fixation   
4177                        0.2377           Fixation   
4179                        0.2399           Fixation   
4180                        0.2480           Fixati

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/user-4-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-4`|
|Recording name| 録画名|`Recording5`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:30:57.621`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:30:57.621`|
|Recording duration| 録画時間|`194658`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`22.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.59`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`79`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.4`mm|
|Average validation precision SD | 平均バリデーション予測SD|`5.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`7.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.26`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.37`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.52`degrees|
|Average validation accuracy |平均バリデーション精度| `61`pixels|
|Average validation precision SD |平均バリデーション予測SD| `18`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `25`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8664         8664            157005027           277678098   
8665         8665            157021668           277694739   
8666         8666            157037711           277710782   
8667         8667            157054951           277728022   
8668         8668            157071592           277744663   
...           ...                  ...                 ...   
10602       10602            189352256           310025327   
10605       10605            189383551           310056622   
10607       10607            189400003           310073074   
10611       10611            189416469           310089540   
10612       10612            189440453           310113524   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8664            293060771.0         566.0         550.0              578.0   
8665            293077411.0         558.0         545.0              575.0   
8666            293093453.0         552.0         554.0              567.0   
8667            293110691.0         554.0         556.0              561.0   
8668            293127331.0         515.0         545.0              517.0   
...                     ...           ...           ...                ...   
10602           325406016.0           NaN           NaN                NaN   
10605           325437309.0           NaN           NaN                NaN   
10607           325453760.0           NaN           NaN                NaN   
10611           325470225.0           NaN           NaN                NaN   
10612           325494208.0           NaN           NaN                NaN   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8664               571.0               554.0               528.0  ...   
8665               577.0               541.0               513.0  ...   
8666               574.0               536.0               534.0  ...   
8667               570.0               546.0               542.0  ...   
8668               554.0               513.0               536.0  ...   
...                  ...                 ...                 ...  ...   
10602                NaN                 NaN                 NaN  ...   
10605                NaN                 NaN                 NaN  ...   
10607                NaN                 NaN                 NaN  ...   
10611                NaN                 NaN                 NaN  ...   
10612                NaN                 NaN                 NaN  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8664                        0.5290                        0.2886   
8665                        0.5340                        0.2816   
8666                        0.5313                        0.2794   
8667                        0.5282                        0.2845   
8668                        0.5128                        0.2673   
...                            ...                           ...   
10602                          NaN                           NaN   
10605                          NaN                           NaN   
10607                          NaN                           NaN   
10611                          NaN                           NaN   
10612                          NaN                           NaN   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8664                         0.4887           Fixation   
8665                         0.4754           Fixation   
8666                         0.4940           Fixation   
8667                         0.5014           Fixation   
8668                         0.4965            Saccade   
...                             ...                ...   
10602                           NaN       EyesNotFound   
10605                           NaN       EyesNotFound   
10607                           NaN       EyesNotFound   
10611                           NaN       EyesNotFo

In [36]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-4/user-4-pic-3.csv")